src/utils/paths.js

```diff
@@ -108,6 +108,18 @@ function pushPathCondition(condition: Value) {
     invariant(left instanceof AbstractValue); // it is a mistake to create an abstract value when concrete value will do
     pushPathCondition(left);
     pushPathCondition(right);
+  } else if (condition.kind === "===") {
+    let [left, right] = condition.args;
+    if (right instanceof AbstractValue && right.kind === "conditional") [left, right] === [right, left];
+    if (left instanceof AbstractValue && left.kind === "conditional") {
+      let [cond, x, y] = left.args;
+      if (right.equals(x) && !right.equals(y)) {
+        pushPathCondition(cond);
```

Comment:
Is this really true? Either `x` and `y` might not be structurally equal to `right`, but that doesn't have to mean that they are not semantically equal.